minio/select_object_reader.py

```diff
@@ -0,0 +1,294 @@
+                    if total_byte_parsed + byte_int(total_byte_length) > read_buffer:
+                        len_read = len(chunked_message[total_byte_parsed:])
+                        message = chunked_message[total_byte_parsed:] + \
+                            self.response.read(byte_int(total_byte_length)-len_read)
+                        end_status = self.decode_message(message, rec)
+                        total_byte_parsed += byte_int(total_byte_length)
+                    # Case 3- the complete message is present in chunked
+                    # messsage.
+                    else:
+                        message = chunked_message[total_byte_parsed:
+                                                  total_byte_parsed +
+                                                  byte_int(total_byte_length)]
+                        total_byte_parsed += byte_int(total_byte_length)
+                        end_status = self.decode_message(message, rec)
+                if end_status:
+                    break
+            return rec
+
+    def extract_header(self, header, header_length):
+        """
+        populates the header map after reading the header
+        """
+        header_map = {}
+        header_byte_parsed = 0
+        # While loop ends when all the headers present are read
+        # header contains multipe headers
+        while header_byte_parsed < header_length:
+            header_name_byte_length = \
+                byte_int(header[header_byte_parsed: header_byte_parsed+1])
+            header_byte_parsed += 1
+            header_name = \
+                header[header_byte_parsed:
+                       header_byte_parsed+header_name_byte_length]
+            header_byte_parsed += header_name_byte_length
+            # Header Value Type is of 1 bytes and is skipped
+            header_byte_parsed += 1
+            value_string_byte_length = \
+                byte_int(header[header_byte_parsed:
+                                header_byte_parsed+2])
+            header_byte_parsed += 2
+            header_value = \
+                header[header_byte_parsed:
+                       header_byte_parsed+value_string_byte_length]
+            header_byte_parsed += value_string_byte_length
+            header_map[header_name.decode("utf-8").lstrip(":")] = \
+                header_value.decode("utf-8").lstrip(":")
+        return header_map
+
+    def read_stats(self, stats):
+        """
+        pupulates the stat dict.
+        """
+        root = cElementTree.fromstring(stats)
+        for attribute in root:
+            if attribute.tag == 'BytesScanned':
+                self.stat['BytesScanned'] = attribute.text
+            elif attribute.tag == 'BytesProcessed':
+                self.stat['BytesProcessed'] = attribute.text
+            elif attribute.tag == 'BytesReturned':
+                self.stat['BytesReturned'] = attribute.text
+
+    def parse_message(self, header_map, payload, payload_length, record):
```

Comment:
All the private functions such as these should have `_` prefix - while they are no explicit private functions. It is customary to use `_` to indicate private functions in python.

Reply:
done